pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1308,6 +1307,9 @@ func (c *PodAffinityChecker) podMatchesPodAffinityTerms(pod, targetPod *v1.Pod,
 	if err != nil {
 		return false, false, err
 	}
+	if targetPodNode == nil {
+		return false, false, fmt.Errorf("node %s was deleted", targetPod.Spec.NodeName)
```

Comment:
We should try to be consistent, in line 1386 below a nil node is not considered an error.